### DISCUSSION TITLE
Valtio with javascript Sets

### DISCUSSION THREAD

**Discussion by Elia-Darwish** [pmndrs/valtio#272] [Q&A] [answered]
Hey friends!

I created a new proxy with one of the properties being a JavaScript `Set` like so
```js
export const selectionState = proxy({
  items: new Set(),
})
```

It is however not triggering re-render when adding or removing from it. Is that normal or am I missing something?
I thought I'd ask here before reporting this as a bug 😄

**Comment by dai-shi** [comment, marked as answer]
Hi, thanks for asking. proxy works for plain objects, and it doesn't track `Set`.
See also #266.

(Now, I think we need to do something with docs...)

**Reply by dai-shi** [reply]
We now have `proxySet` and `proxyMap` utils!
They emulate `Set` and `Map` behaviors, which should work for typical use cases.
Related discussion: https://github.com/pmndrs/valtio/discussions/114#discussioncomment-1982749
